lib/paper_trail/model_config.rb

```diff
@@ -144,22 +144,35 @@ def setup_associations(options)
       # In PR _____ () we ask them to use `paper_trail_options` instead.
       @model_class.class_attribute :version_class_name
       @model_class.version_class_name = options[:class_name] || "PaperTrail::Version"
+      assert_concrete_activerecord_class(@model_class.version_class_name)
+
+      setup_has_many_versions(options)
+    end
+
+    def setup_has_many_versions(options)
+      unless options[:versions].is_a?(Hash)
+        options[:versions] = {
+          name: options[:versions]
+        }
+      end
 
       # @api private - versions_association_name
       @model_class.class_attribute :versions_association_name
-      @model_class.versions_association_name = options[:versions] || :versions
+      @model_class.versions_association_name = options[:versions].delete(:name) || :versions
+
+      scope = options[:versions].delete(:scope) || -> { order(model.timestamp_sort_order) }
+      options[:versions].assert_valid_keys(valid_keys_for_has_many)
```

Comment:
Neat! I did not know about `assert_valid_keys`.

What would happen if we didn't validate keys? Would `has_many` complain? I think I'd rather let `has_many` validate these, if it does.

Reply:
Works for me! Not sure why I didn't think of that. :)

And yep, `has_many` raises the exact same error (since it uses `assert_valid_keys` too—that's where I learned about it)... so I don't even have to change my test. :)